doc/data_protocol.md

```diff
@@ -86,33 +86,90 @@ The length of the data field is variable
+
 # Data protocol
 
-The term `ID` is used below to denote `identifier field` of the CAN Bus and byte 2 (`Frame ID`) of a frame over the serial links
+The term `ID` is used below to denote `identifier field` of the CAN Bus and byte 2 (`Frame ID`) of a frame over the serial links.
 
-The `identifier field` is an 11 bits field but only the 8 lower bits are used to make the data on the CAN Bus as similar as possible to the data on the serial buses. The remaining `ID` values `100` - `7FF` are available for data transfer between on-board systems over the CAN Bus only and cannot be sent to the ground.
+The `identifier field` is an 11 bits field but only the 8 lower bits are used to make the data on the CAN Bus as similar as possible to the data on the serial buses. The remaining `ID` values `100` - `7FF` are available for data transfer between on-board systems over the CAN Bus only, and cannot be sent to ground.
 
-Each message on the Telecommand link and the Telemetry link has a unique ID
+Each message on the Telecommand TC link and the Telemetry TM link has a unique ID.
 
-| ID              | Frame type | From                        | To                          |
-| --------------- | ---------- | --------------------------- | --------------------------- |
-| `0x00` - `0x0F` | TC         | Ground or Flight Controller | Engine Computer             |
-| `0x10` - `0x7F` | TM         | Engine Computer             | Ground or Flight Controller |
-| `0x80` - `0x8F` | TC         | Ground                      | Flight Controller           |
-| `0x90` - `0xEF` | TM         | Flight Controller           | Ground                      |
-| `0xF0` - `0xFF` | N/A        | N/A                         | N/A                         |
+<table>
+<thead>
+  <tr>
+    <th>ID</th>
+    <th>Frame Type</th>
+    <th>From</th>
+    <th>To</th>
+    <th>Unique IDs</th>
+  </tr>
+</thead>
+<tbody>
+  <tr>
+    <td>0x00 - 0x0F</td>
+    <td>TC</td>
+    <td>Ground Station</td>
+    <td>Flight Controller</td>
+    <td>16</td>
+  </tr>
+  <tr>
+    <td>0x10 - 0x1F</td>
+    <td>TC</td>
+    <td>Flight Controller</td>
+    <td>Ground Station</td>
+    <td>16</td>
+  </tr>
+  <tr>
+    <td>0x20 - 0x2F</td>
+    <td>TC</td>
+    <td>Ground Station</td>
+    <td>Engine Computer</td>
+    <td>16</td>
+  </tr>
+  <tr>
+    <td>0x30 - 0x3F</td>
+    <td>TC</td>
+    <td>Engine Computer</td>
+    <td>Ground Station</td>
+    <td>16</td>
+  </tr>
+  <tr>
+    <td>0x40 - 0x7F</td>
+    <td>TM</td>
+    <td>Flight Controller</td>
+    <td>Ground Station</td>
+    <td>64</td>
+  </tr>
+  <tr>
+    <td>0x80 - 0xBF</td>
+    <td>TM</td>
+    <td>Engine Computer</td>
+    <td>Ground Station</td>
+    <td>64</td>
+  </tr>
+  <tr>
+    <td>0xC0 - 0xFF</td>
+    <td>N/A</td>
+    <td>N/A</td>
+    <td>N/A</td>
+    <td>64</td>
+  </tr>
+</tbody>
+</table>
```

Comment:
I'm quite skeptical to strictly defining these ranges at this point in time.

Previously, we had allocated 111 message IDs for Edda to send to any receiver, now we're only allowing 80 IDs. Is there a reason for this change?

Reply:
I have choose the ranges according to the different data we want to transmit, we still have plenty of spare IDs for each pair of sender/receiver. There is still 64 spare IDs just next to the EC to GS telemetry that we could use if we need more in the future.